caddy/caddy.go

```diff
@@ -0,0 +1,152 @@
+// Package caddy provides a handler for Caddy Server (https://caddyserver.com/)
+// allowing to turn any web API in a one supporting the Vulcain protocol.
+package caddy
+
+import (
+	"bytes"
+	"net/http"
+	"strconv"
+	"sync"
+
+	"github.com/caddyserver/caddy/v2"
+	"github.com/caddyserver/caddy/v2/caddyconfig/caddyfile"
+	"github.com/caddyserver/caddy/v2/caddyconfig/httpcaddyfile"
+	"github.com/caddyserver/caddy/v2/modules/caddyhttp"
+	"github.com/dunglas/vulcain"
+	"go.uber.org/zap"
+)
+
+func init() {
+	caddy.RegisterModule(Vulcain{})
+	httpcaddyfile.RegisterHandlerDirective("vulcain", parseCaddyfile)
+}
+
+var bufPool = sync.Pool{
+	New: func() interface{} {
+		return new(bytes.Buffer)
+	},
+}
+
+type Vulcain struct {
+	OpenAPIFile string `json:"openapi_file,omitempty"`
+	MaxPushes   int    `json:"max_pushes,omitempty"`
```

Comment:
Make sure to add Godocs here so the fields get automatic documentation on the Caddy website!